base/twiceprecision.jl

```diff
@@ -456,6 +456,9 @@ step(r::StepRangeLen{T,TwicePrecision{T},TwicePrecision{T}}) where {T} = T(r.ste
 range_start_step_length(a, st::IEEEFloat, len::Integer) =
     range_start_step_length(oftype(st, a), st, len)
 
+range_start_step_length(a::IEEEFloat, st, len::Integer) =
+    range_start_step_length(a, oftype(a, st), len)
```

Comment:
We might want to use `promote` for these?
```suggestion
    range_start_step_length(promote(a, st)..., len)
```